src/components/PresenceMap/component.js

```diff
@@ -0,0 +1,194 @@
+import React, { useRef, useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import getMapContainerStyles from './getMapContainerStyles';
+import StyledContainer from './styled/container';
+import StyledMapContainer from './styled/mapContainer';
+
+const MAX_ZOOM = 14;
+
+export default function MapComponent(props) {
+    const { showTraffic, showTransit, center, zoom, objects, style } = props;
+    const container = useRef();
+    const mapContainer = useRef();
+    const [map, setMap] = useState(null);
+
+    const prevIsScriptLoaded = useRef(props.isScriptLoaded);
+    useEffect(() => {
+        if (!prevIsScriptLoaded.current && props.isScriptLoaded && props.isScriptLoadSucceed) {
```

Comment:
this is a common pattern (checking for prev prop values), in order to reuse we can create a hook:
```
function usePrevious(value) {
  const ref = useRef();
  useEffect(() => {
    ref.current = value;
  });
  return ref.current;
}
```
and put it in:
```src/libs/hooks/index.js```
then it will be used here as:
```const prevIsScriptLoaded = usePrevious(props.isScriptLoaded);```